helios-services/src/main/java/com/spotify/helios/agent/AgentZooKeeperRegistrar.java

```diff
@@ -53,13 +56,19 @@
   private final Service agentService;
   private final String name;
   private final String id;
+  private final long zooKeeperRegistrationTtlMillis;
+  private Clock clock;
 
   private PersistentEphemeralNode upNode;
 
-  public AgentZooKeeperRegistrar(final Service agentService, final String name, final String id) {
+  public AgentZooKeeperRegistrar(final Service agentService, final String name, final String id,
+                                 final int zooKeeperRegistrationTtlMinutes) {
     this.agentService = agentService;
     this.name = name;
     this.id = id;
+    this.zooKeeperRegistrationTtlMillis =
+        TimeUnit.MILLISECONDS.convert(zooKeeperRegistrationTtlMinutes, TimeUnit.MINUTES);
+    this.clock = new SystemClock();
```

Comment:
Can you not test `AgentZooKeeperRegistrar` in isolation, without CLI and crap?

At a glance, `agentService` seems easily mockable (only `stopAsync()` is called). `name` and `id` should be as well.